go/integTest/pubsub_basic_test.go

```diff
@@ -3,290 +3,964 @@
+			channelNames:   []string{"test-channel-1", "test-channel-2", "test-channel-3"},
+			messageContent: "test message",
+		},
+		{
+			name:           "Cluster with SyncLoop",
+			clientType:     GlideClusterClient,
+			readMethod:     SyncLoopMethod,
+			useCallback:    false,
+			channelNames:   []string{"test-channel-1", "test-channel-2", "test-channel-3"},
+			messageContent: "test message",
+		},
 	}
 
-	// Verify results
-	assert.Equal(suite.T(), messageCount, len(received1), "Client 1 should receive exactly %d messages", messageCount)
-	assert.Equal(suite.T(), messageCount, len(received2), "Client 2 should receive exactly %d messages", messageCount)
-
-	// Verify message content and order
-	for i := 0; i < messageCount; i++ {
-		if i < len(received1) {
-			expectedMsg1 := fmt.Sprintf("Hello Subscribers - msg %d", i)
-			assert.Equal(suite.T(), expectedMsg1, received1[i].Message, "Unexpected message for client 1")
-		}
-
-		if i < len(received2) {
-			expectedMsg2 := fmt.Sprintf("Hello Subscribers - msg %d", i)
-			assert.Equal(suite.T(), expectedMsg2, received2[i].Message, "Unexpected message for client 2")
-		}
-	}
-}
+	for _, tt := range tests {
+		suite.T().Run(tt.name, func(t *testing.T) {
+			publisher := suite.createAnyClient(tt.clientType, nil)
 
-// TestUnsubscribeWithGlideClient tests unsubscribe functionality
-func (suite *PubSubTestSuite) TestUnsubscribeWithGlideClient() {
-	suite.T().Skip("Skipping unsubscribe test due to Rust implementation limitation - not yet implemented in lib.rs:574")
+			// Create channel definitions for all channels
+			channels := make([]ChannelDefn, len(tt.channelNames))
+			for i, channelName := range tt.channelNames {
+				channels[i] = ChannelDefn{Channel: channelName, Mode: ExactMode}
+			}
 
-	// Create a publisher client
-	publisher := suite.createDefaultClient("publisher")
+			// Create expected messages map
+			expectedMessages := make(map[string]string)
+			for _, channelName := range tt.channelNames {
+				expectedMessages[channelName] = tt.messageContent
+			}
 
-	// Create a subscriber client with subscription configuration
-	messageReceived := make(chan *api.PubSubMessage, 10)
-	callback := func(message *api.PubSubMessage, ctx any) {
-		messageReceived <- message
-	}
+			var receiver api.BaseClient
+			queues := make(map[int]*api.PubSubMessageQueue)
+			if !tt.useCallback {
+				receiver = suite.CreatePubSubReceiver(tt.clientType, channels, 1, false)
+				queue, err := receiver.GetQueue()
+				assert.Nil(t, err)
+				queues[1] = queue
+			} else {
+				receiver = suite.CreatePubSubReceiver(tt.clientType, channels, 1, true)
+			}
 
-	subscriptionConfig := api.NewStandaloneSubscriptionConfig().
-		WithSubscription(api.ExactChannelMode, "test-channel-unsub").
-		WithCallback(callback, nil)
+			// Allow subscription to establish
+			time.Sleep(100 * time.Millisecond)
+
+			// TODO: For SignalChannel Async tests this will result in all of the messages getting stacked in the queue
```

Comment:
Do you want to complete this todo in this PR? If no, please move to the MD file you have.